latest version of jasper causes build to break

With jasper-3.0.3:

```
[ 47%] Building C object CMakeFiles/g2c.dir/src/dec_jpeg2000.c.o
In file included from /usr/local/jasper-3.0.3/include/jasper/jasper.h:73,
                 from /home/ed/NCEPLIBS-g2c/src/dec_jpeg2000.c:10:
/usr/local/jasper-3.0.3/include/jasper/jas_config.h:115:2: warning: #warning "Your code is being built against an older version of the C standard than JasPer was.  Although this is supported, this may require some extra preprocessor defines when building." [-Wcpp]
  115 | #warning "Your code is being built against an older version of the C standard than JasPer was.  Although this is supported, this may require some extra preprocessor defines when building."
      |  ^~~~~~~
[ 48%] Building C object CMakeFiles/g2c.dir/src/enc_jpeg2000.c.o
In file included from /usr/local/jasper-3.0.3/include/jasper/jasper.h:73,
                 from /home/ed/NCEPLIBS-g2c/src/enc_jpeg2000.c:9:
/usr/local/jasper-3.0.3/include/jasper/jas_config.h:115:2: warning: #warning "Your code is being built against an older version of the C standard than JasPer was.  Although this is supported, this may require some extra preprocessor defines when building." [-Wcpp]
  115 | #warning "Your code is being built against an older version of the C standard than JasPer was.  Although this is supported, this may require some extra preprocessor defines when building."
      |  ^~~~~~~
[ 50%] Building C object CMakeFiles/g2c.dir/src/jpcpack.c.o
[ 51%] Building C object CMakeFiles/g2c.dir/src/jpcunpack.c.o
[ 52%] Linking C static library libg2c.a
[ 52%] Built target g2c
[ 53%] Building C object tests/CMakeFiles/tst_addfield3.dir/tst_addfield3.c.o
[ 54%] Linking C executable tst_addfield3
/usr/bin/ld: ../libg2c.a(dec_jpeg2000.c.o): in function `dec_jpeg2000':
/home/ed/NCEPLIBS-g2c/src/dec_jpeg2000.c:46: undefined reference to `jpc_decode'
/usr/bin/ld: ../libg2c.a(enc_jpeg2000.c.o): in function `enc_jpeg2000':
/home/ed/NCEPLIBS-g2c/src/enc_jpeg2000.c:104: undefined reference to `jpc_encode'
collect2: error: ld returned 1 exit status
make[2]: *** [tests/CMakeFiles/tst_addfield3.dir/build.make:89: tests/tst_addfield3] Error 1
make[1]: *** [CMakeFiles/Makefile2:968: tests/CMakeFiles/tst_addfield3.dir/all] Error 2
make: *** [Makefile:141: all] Error 2
ed@mikado:~/NCEPLIBS-g2c/b$ 
```

--- a/.github/workflows/linux.yml
+++ b/.github/workflows/linux.yml
@@ -9,7 +9,7 @@ jobs:
       CXX: g++
     strategy:
       matrix:
-        jasper-version: [2.0.33 3.0.3]
+        jasper-version: [2.0.33, 3.0.3]
     
     steps:
     - name: install-dependencies

--- a/.github/workflows/linux.yml
+++ b/.github/workflows/linux.yml
@@ -9,7 +9,7 @@ jobs:
       CXX: g++
     strategy:
       matrix:
-        jasper-version: [2.0.33, 3.0.3]
+        jasper-version: [2.0.33]
     
     steps:
     - name: install-dependencies

--- a/.github/workflows/linux.yml
+++ b/.github/workflows/linux.yml
@@ -9,7 +9,7 @@ jobs:
       CXX: g++
     strategy:
       matrix:
-        jasper-version: [3.0.3]
+        jasper-version: [2.0.33 3.0.3]
     
     steps:
     - name: install-dependencies

--- a/.github/workflows/linux.yml
+++ b/.github/workflows/linux.yml
@@ -7,7 +7,7 @@ jobs:
     env:
       CC: gcc
       CXX: g++
-    stragegy:
+    strategy:
       matrix:
         jasper-version: [3.0.3]
     

--- a/.github/workflows/linux.yml
+++ b/.github/workflows/linux.yml
@@ -9,7 +9,7 @@ jobs:
       CXX: g++
     strategy:
       matrix:
-        jasper-version: [2.0.33]
+        jasper-version: [2.0.33, 3.0.3]
     
     steps:
     - name: install-dependencies

--- a/.github/workflows/linux.yml
+++ b/.github/workflows/linux.yml
@@ -7,7 +7,10 @@ jobs:
     env:
       CC: gcc
       CXX: g++
-
+    stragegy:
+      matrix:
+        jasper-version: [3.0.3]
+    
     steps:
     - name: install-dependencies
       run: |
@@ -20,7 +23,7 @@ jobs:
       with:
         repository: jasper-software/jasper
         path: jasper
-        ref: version-3.0.3
+        ref: version-${{ matrix.jasper-version }}
 
     - name: cache-jasper
       id: cache-jasper

--- a/.github/workflows/linux.yml
+++ b/.github/workflows/linux.yml
@@ -20,7 +20,7 @@ jobs:
       with:
         repository: jasper-software/jasper
         path: jasper
-        ref: version-2.0.25
+        ref: version-3.0.3
 
     - name: cache-jasper
       id: cache-jasper

